client/src/handlers/handlers.ts

```diff
@@ -14,6 +14,86 @@ export function makeHandleRunJob (
   }
 }
 
+let flixFileLocation
+const FLIX_GLOB_PATTERN = '**/*.flix'
+
+export function setFlixFileName(Filename:String) {
+    flixFileLocation = Filename
+}
+
+function selectTerminal(): Thenable<vscode.Terminal | undefined> {
+	interface TerminalQuickPickItem extends vscode.QuickPickItem {
+		terminal: vscode.Terminal;
+	}
+	const terminals = <vscode.Terminal[]>(<any>vscode.window).terminals;
+	let items: TerminalQuickPickItem[] = terminals.map(t => {
+		return {
+			label: `name: ${t.name}`,
+			terminal: t
+		};
+	});
+    items.push({label:`create a new terminal`, terminal: undefined})
+	return vscode.window.showQuickPick(items).then(item => {
+		return item ? item.terminal : undefined;
+	});
+}
+
+function findFlixTerminal() {
+    const activeTerminals = vscode.window.terminals
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        if(element.name == `flix`)
+            return element
+    }
+    return undefined
+}
+
+function showActiveTerminals() {
+    const activeTerminals = vscode.window.terminals
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        console.log(element)
+    }
+}
+
+export async function compileInTerminal() {
+    
+    const files = await vscode.workspace.findFiles(FLIX_GLOB_PATTERN)
+    let cmd = "java -jar "+flixFileLocation;
```

Comment:
Need to ensure that all paths are escaped correctly. Maybe there is a package or function in VSCode?